packages/sdk-rtl/src/paginator.ts

```diff
@@ -0,0 +1,325 @@
+  /** Captured from the original pagination request */
+  options?: Partial<ITransportSettings>
+  /** Total number of pages. -1 if not known. */
+  pages: number
+  /** Current page. -1 if not known. */
+  page: number
+
+  /**
+   * Is the specified link defined for this Link header?
+   * @param link to check
+   */
+  hasRel(link: PageLinkRel): boolean
+
+  /** Get the requested relative link
+   * if the requested link is not defined, all calculated values are reset, including
+   * `total`, `items`, `offset`, and `limit`
+   */
+  getRel(link: PageLinkRel): Promise<SDKResponse<TSuccess, TError>>
+  /** Get the first page of items. This is the same as offset=0 */
+  firstPage(): Promise<SDKResponse<TSuccess, TError>>
+  /**
+   * Get the last page of items
+   *
+   * @remarks This link is only provided if `total` is known.
+   */
+  lastPage(): Promise<SDKResponse<TSuccess, TError>>
+  /**
+   * Get the next page of items
+   *
+   * @remarks This link is provided if `total` is known, or if the number of items returned == `limit`. In the latter case, this function may return an empty result set.
+   */
+  nextPage(): Promise<SDKResponse<TSuccess, TError>>
+  /**
+   * Get the previous page of items
+   *
+   * @remarks This link is provided if the last page was not the first page.
+   */
+  prevPage(): Promise<SDKResponse<TSuccess, TError>>
+}
+
+/** Pagination function call */
+export type PaginateFunc<TSuccess, TError> = () => Promise<
+  SDKResponse<TSuccess, TError>
+>
+
+/**
+ * Parse a link header to extract rels
+ * @param linkHeader to parse
+ *
+ * Several different approaches are discussed at https://stackoverflow.com/questions/8735792/how-to-parse-link-header-from-github-api
+ *
+ * Unfortunately, none of them are a good implementation
+ *
+ */
+export const linkHeaderParser = (linkHeader: string): PageLinks => {
+  const re = /<\s*(.*)\s*>;\s*rel="\s*(.*)\s*"\s*/gm
+  const links = linkHeader.split(',')
+  const obj: PageLinks = {}
+  let arrRes
+
+  links.forEach((link) => {
+    link = link.trim()
+    while ((arrRes = re.exec(link))) {
+      const key = arrRes[2].split(' ')[0].trim().toLocaleLowerCase()
+      obj[key] = {
+        url: arrRes[1].trim(),
+        rel: key as PageLinkRel,
+        name: arrRes[2].trim(),
+      }
+    }
+  })
+  return obj
+}
+
+/**
+ * Create an API paginator for an endpoint that returns a Link header
+ * @param sdk implementation of IAPIMethods. Can be full SDK or functional auth session
+ * @param func sdk call that includes a pagination header
+ * @param options transport options override to capture and use in paging requests
+ *
+ * @example
+ * ```ts
+ * const paged = new paginate(sdk, (sdk) => search_dashboards(sdk, { limit: 10 })
+ * const dashboards = paged.items
+ * while (paged.has('next')) {
+ *   await paged.next()
+ *   dashboards.push(paged.items)
+ * }
+ * console.log(`${paged.total} dashboards retrieved`)
+ * ... (some code to list all dashboards )
+ *
+ * ```
+ */
```

Comment:
The example snippet is outdated and doesn't have the `nextPage()` rename but this is a pattern for results a page at a time. The `next` iterator hasn't been implemented yet.

Reply:
I moved this example to docs/pagination.md and we'll verify it when the functional tests are completed